# Seurat

This is a chapter includes example using Seurat to analyze single cell RNA sequencing data. 

## The famous pbmc3k

This is the first chapter in Seurat tutorial. You can find the full description [here](https://bit.ly/31G8iXw).  
**Data: ** 2,700 Periperal Blood Mononuclear Cells (PBMCs) freely from 10X Genomics.  
**Sequencing techniques: ** Illumina NextSeq500 with UMI  
**Data Sturcture: ** gene in row and cell in column.

```{r import 2700 PBMCs data, message=FALSE, warning=FALSE}
library(dplyr)
library(Seurat)
library(patchwork)

# Load the PBMC data
pbmc.data <- Read10X(data.dir = "rawData/filtered_gene_bc_matrices/hg19/")

# Initialize the Seurat object with the raw (non-normalized data)
pbmc <- CreateSeuratObject(
  counts = pbmc.data,
  project = "pbmc3k",
  min.cells = 3,
  min.features = 200
)

pbmc
```

### Standard process {-}  
**1.QC and selecting cells for further analysis**  
**2.Data normalization**  
**3.Detection of highly variable features**  
**4.Scaling the data**

1.QC filtering:
```{r}
# The [[ operator can add columns to object metadata. This is a great place to stash QC stats
# Calculate mitochondrial gene using the start of MT using PercentageFeatureSet()
# Low-quality/dying cells often exhibit extensive Mitochondrial contamination
pbmc[["percent.mt"]] <- PercentageFeatureSet(pbmc, pattern = "^MT-")
head(pbmc@meta.data, 5)
```

Visualize the QC metrics and we use these to filter our cells.  
- We filter cells that have unique feature counts **> 2500** or **< 200**.  
- We filter cells that have **> 5 %** mitochondrial counts.  
The function for subset is `subset()`
```{r}
# Visualize QC metrics as a violin plot
VlnPlot(pbmc, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
```

```{r}
# FeatureScatter is typically used to visualize feature-feature relationships, but can be used
# for anything calculated by the object, i.e. columns in object metadata, PC scores etc.

plot1 <- FeatureScatter(pbmc, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(pbmc, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
plot1 + plot2
```

```{r}
pbmc <- subset(pbmc, subset = nFeature_RNA > 200 & nFeature_RNA < 2500 & percent.mt < 5)
```

2.Normalizing the data  
Method: **LogNormalize**, normalizes the feature expression measurements for each cell by the total expression and then multiplies by a scale factor, defaut is 10,000. Then use log-transforms the result. Result are stored in `pbmc[["RNA]]@data`
```{r}
pbmc <- NormalizeData(pbmc, normalization.method = "LogNormalize", scale.factor = 10000)
```

3.Feature Selection  
We want to find if there is sone features occur in cell-to-cell (high cell-to-cell variation)
```{r, warning=FALSE}
# select 2000 Features
pbmc <- FindVariableFeatures(pbmc, selection.method = "vst", nfeatures = 2000)

# Identify the 10 most highly variable genes
top10 <- head(VariableFeatures(pbmc), 10)

# plot variable features with and without labels
plot1 <- VariableFeaturePlot(pbmc)
plot2 <- LabelPoints(plot = plot1, points = top10, repel = TRUE)
plot1 + plot2
```

4. Scaling the data `ScaleData()` to perform linear dimensional reduction(PCA)   
The result is stored in `pbmc[["RNA"]]@scale.daa`
```{r}
# make mean expression of each gene across cells is 0
# scale expression of each gene, so the variation across cells is 1, provide equal weight in downstream analysus so that highly-expressed genes do not dominate
all.genes <- rownames(pbmc)
pbmc <- ScaleData(pbmc, features = all.genes)

# run PCA
pbmc <- RunPCA(pbmc, features = VariableFeatures(object = pbmc))

# visualize, VizDimReduction(), DimPlot(), DimHeatmap()
VizDimLoadings(pbmc, dims = 1:2, reduction = "pca")
DimPlot(pbmc, reduction = "pca")
DimHeatmap(pbmc, dims = 1:3, cells = 500, balanced = TRUE)
```
After these work, it is time for us to determine the dimensionality of the dataset.
